src/metabase/events/schema.clj

```diff
@@ -4,6 +4,14 @@
    [malli.util :as mut]
    [toucan2.core :as t2]))
 
+;; collection events
+(let [default-schema (mc/schema
+                      [:map {:closed true}
+                       [:user-id  pos-int?]
+                       [:object   [:fn #(t2/instance-of? :model/Collection %)]]])]
```

Comment:
I think it's cargo culted from some more complex cases that need de-duplication, like [this one](https://github.com/metabase/metabase/blob/325b9827b446f3bbd6000257b21a3ece8bb17d03/src/metabase/events/schema.clj#L37-L50).

Reply:
yeah my point was that those cases could be like this:
```
(def ^:private card-events-schemas
    (let [default-schema (mc/schema
                          [:map {:closed true}
                           [:user-id  [:maybe pos-int?]]
                           [:object   [:fn #(t2/instance-of? :model/Card %)]]])]
      {:event/card-create default-schema
       :event/card-read   default-schema
       :event/card-update default-schema
       :event/card-delete default-schema
       :event/card-query  [:map {:closed true}
                           [:card-id pos-int?]
                           [:user-id [:maybe pos-int?]]
                           [:context {:optional true} :any]]}))
```

Reply:
Nope, they were [added like this](https://github.com/metabase/metabase/pull/34947/files) - maybe the reviewer knows more 🤷 😉

Reply:
I wrote those. I used let because things like `defualt-schema` is only temporarily needed to build the actual schema.

it makes more sense when you have more event types tho.

like

```clojure
(let [default-schema (mc/schema
                      [:map {:closed true}
                       [:user-id  pos-int?]
                       [:object   [:fn #(t2/instance-of? :model/Card %)]]])]
  (def ^:private card-events-schemas
    {:event/card-create default-schema
     :event/card-read   default-schema
     :event/card-update default-schema
     :event/card-delete default-schema
     :event/card-query  [:map {:closed true}
                         [:card-id pos-int?]
                         [:user-id [:maybe pos-int?]]
                         [:context {:optional true} :any]]}))
```

Reply:
Just to make sure I understand the rationale - is it so that you see the shape of the data directly inside the def, i.e. factors out some noise? Personally I'd prefer to keep the def on the outside, I expect vars to get defined without indentation and this makes it a bit harder to "scan" the namespace, but now I notice how it's nice to see the map's shape directly below if you go-to-definition. Haskell's "where" notation would give us best of both 😄 

Thanks for indulging my archeological curiosity.